web/app/schedules/home/page.tsx

```diff
@@ -13,6 +13,34 @@ import ClickOutsideHandler from './handlers/ClickOutsideHandler';
 
 import ShowPopUpContentContextProvider from '@/app/contexts/ShowPopUpContentContext';
 
+import { ClassValueType } from '@/app/contexts/SelectedClassesContext/types';
+
+function DisciplineBlockJSX({ ...props }: {
+    index: number,
+    cls: ClassValueType,
+    classesToShow: ClassValueType[]
+}) {
+    let classAppearance;
+    for (let indexInner = 0; indexInner < props.classesToShow.length; indexInner++) {
+        if (props.classesToShow[indexInner].discipline.code === props.cls.discipline.code) {
+            classAppearance = indexInner;
+            break;
+        }
+    }
+
+    return (
+        <Fragment key={props.index}>
+            {props.index === classAppearance ?
+                <span className='flex justify-center items-center rounded-md font-medium bg-primary bg-opacity-80 w-10/12 h-8'>
+                    {props.cls.discipline.name}
+                </span>
+                : null
+            }
```

Comment:
Apenas e exclusivamente para o quesito aparência, sugiro o uso do operador `&&`. 
Fonte: [Logical AND operator (&&)](https://react.dev/learn/conditional-rendering#logical-and-operator-)